src/api.js

```diff
@@ -662,6 +663,7 @@ api.declare({
     var task = await this.Task.create({
       taskId:             taskId,
       provisionerId:      taskDef.provisionerId,
+      clientId:           taskDef.clientId === clientId && taskDef.clientId || clientId, 
```

Comment:
I think it would be easier -- and clearer -- to just use `clientId` here.  That will *always* be the correct value.

As written, this is saying "x if x is the same as y, otherwise y", which is a long way of saying "y" :)

Also, if the user provides an incorrect clientId, this conditional will just ignore it, rather than returning an error.